experimental/Schemes/CoherentSheaves.jl

```diff
@@ -761,9 +832,76 @@ underlying_presheaf(M::HomSheaf) = M.M
 sheaf_of_rings(M::HomSheaf) = M.OOX
 domain(M::HomSheaf) = M.domain
 codomain(M::HomSheaf) = M.codomain
+#default_covering(M::HomSheaf) = default_covering(domain(M)) # TODO: This is only a temporary fix!
+
+########################################################################
+# Sheaves of direct sums                                               #
+########################################################################
+@attributes mutable struct DirectSumSheaf{SpaceType, OpenType, OutputType,
+                                          RestrictionType
+                                         } <: AbsCoherentSheaf{
+                                                               SpaceType, OpenType,
+                                                               OutputType, RestrictionType
+                                                              }
+  summands::Vector{AbsCoherentSheaf{SpaceType, OpenType, OutputType, RestrictionType}}
+  #projections::Vector #TODO: Realize as sections in HomSheafs
+  #inclusions::Vector # TODO: same.
+  OOX::StructureSheafOfRings
+  M::PreSheafOnScheme
+
+  function DirectSumSheaf(X::AbsCoveredScheme, summands::Vector{<:AbsCoherentSheaf})
+    all(x->(scheme(x)===X), summands) || error("summands must be defined over the same scheme")
+    OOX = OO(X)
+    all(x->(sheaf_of_rings(x)===OOX), summands) || error("summands must be defined over the same sheaves of rings")
+
+    ### Production of the modules on open sets; to be cached
+    function production_func(FF::AbsPreSheaf, U::AbsSpec)
+      result, inc, pr = direct_sum([F(U) for F in summands]..., task=:both)
+      set_attribute!(result, :inclusions, inc) # TODO: Workaround as long as the maps are not cached.
+      set_attribute!(result, :projections, pr) 
```

Comment:
@jankoboehm :
As already indicated in my mail, I didn't find directly how to get my hands on the canonical projections and inclusions of the summands via the caching of canonical maps. If this can already be solved in another way, please let me know! 

Otherwise, this seems to do the job for the time being. So working everything else out until Monday does not seem to be that urgent anymore.